### ISSUE TITLE
recent trouble with browser.search("geneSymbol") (no trouble with 1.0.9)

### ISSUE THREAD

**Issue by paul-shannon** [igvteam/igv.js#481]
![image](https://user-images.githubusercontent.com/2480712/32859653-f6b4b75e-ca03-11e7-9c11-b5d017b1a502.png)

I cannot reproduce this in a minimal html file, where search always works, but it happens regularly now when I build from the full npm igv.js - either at npm, or from my own freshly updated igv.js.npm github repo.  In all cases, a chromosomal location ("chrN:start-end") works fine.

When I switch from awebpack to an old-fashioned use  of:
     <script src="http://igv.org/web/release/1.0.9/igv-1.0.9.js"></script>
     <link rel="stylesheet" type="text/css" href="http://igv.org/web/release/1.0.9/igv-1.0.9.css">
search works again.  It works with today's beta release too.  Suggesting that there might be a weird interaction in the new npm/webpack version only.

Before I try to distill a minimal, non-npm reproduction of this problem, I thought I'd check in to see if this sounds at all familiar to you.

My mm10 options are shown below.  I have to admit, also, that though the documentation says that a searchable track (Gencode vM14 below) should not be indexed, I believe I see that
 - an indexURL is needed in order to load the annotation file incrementally
 - that it is indexed does not seem to interfere with searchability, with 1.0.9 and the beta build, when I include igv.js with a script tag.   There are clearly things I don't understand about using a searchable reference.
  - should I use a mimimal, small-file, gene-coordinates-only file as the search track, following all the recommended options?  Perhaps I am playing with fire by using a full gtf file for the search track.
 
````
 var mm10_options = {
         flanking: 2000,
	 showKaryo: false,
         showNavigation: true,
         minimumBases: 5,
         showRuler: true,
         reference: {id: "mm10",
                     fastaURL: "http://trena.systemsbiology.net/mm10/GRCm38.primary_assembly.genome.fa",
                     cytobandURL: "http://trena.systemsbiology.net/mm10/cytoBand.txt"
                     },
         tracks: [
            {name: 'Gencode vM14',
             url: "http://trena.systemsbiology.net/mm10/gencode.vM14.basic.annotation.sorted.gtf.gz",
             indexURL: "http://trena.systemsbiology.net/mm10/gencode.vM14.basic.annotation.sorted.gtf.gz.tbi",
             format: 'gtf',
             visibilityWindow: 2000000,
             displayMode: 'EXPANDED'
             },
            ]
       }; // mm10_options

````

Thanks.

**Comment by jrobinso** [comment]
Could you try using these:  http://igv.org/web/test/dist/

"beta" is updated periodically,  but has not been updated since prior to
the npm release.   If you can reproduce the problem with the versions above
it will eliminate npm as the cause.

We run a search webservice for a few genomes, I think mm10 is one of them,
and it will be tried if you don't have any searchable tracks.   Indexed
tracks cannot work in general because it searches through features it has
loaded.   If it hasn't loaded the feature you are searching for it won't
find it, of course.   The point of indexing is to prevent loading features
that are not in view.

I think you have to explicitly make a feature track searchable,  check the
documentation,  but the point is mute if the feature file is indexed.

**Comment by paul-shannon** [comment]
Hi Jim,

I tried the web/test igv.js and igv.css, using a <script> tag to load.  Alas, I see the same problem.  If you have further ideas for things I can try, please send them along.  

I see now that I have been using your search webservices for hg38 and mm10 all along, without realizing it.  Your comments about the unworkability of index tracks makes perfect sense.

 - Paul

**Comment by jrobinso** [comment]
Paul, unzip and try the attached example file and see if search works.  It works for me.   If it does work for you we have to figure out what's different in your example html vs this html.   If it doesn't work we have to figure out what's different in our environments.  
[mm10.html.zip](https://github.com/igvteam/igv.js/files/1476744/mm10.html.zip)

**Comment by paul-shannon** [comment]
Jim, this is helpful.  

I made one small change to your example, assigning a global variable for easy access in the javascript console:

    igvBrowser = igv.createBrowser(div, options);

Then the contrast is clear:

  success: if I type “tcf7” into the search box 
  success: if I type "chr11:94,934,223-94,955,042” into the search box
  success: from the console: igvBrowser.search("chr11:94,934,223-94,955,042”)
  failure: from the console: igvBrowser.search("tcf7”)
 
Do you see the same thing?

 - Paul

**Comment by jrobinso** [comment]
Yes, I can reproduce it now.  It is a bug.   We'll fix it soon.

**Comment by jrobinso** [comment]
Bug is fixed.   However, @turner   browser.search() should use the same code internally as the "go" box, and take the same arguments.

**Comment by jrobinso** [comment]
@paul-shannon update from npm again and see if the problem is fixed (0.1.1).   BTW The NPM version numbers reflect the fact that we haven't released an igv.js NPM version yet, by that I mean that igv.js itself is not officially released.   When it is NPM and igv.js versions will be in sync.
